### PR TITLE
BUG: Fix DataFrame.apply(..., raw=True) not calling with raw array

### PR DESCRIPTION
**Pull request by kernc**
- [x] closes https://github.com/pandas-dev/pandas/issues/32423
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by kernc** [comment]
Seems to pass. :tada: Where do I put the whatsnew?

File: pandas/tests/frame/test_apply.py [hunk 1]
```diff
@@ -235,7 +235,7 @@ def test_apply_broadcast_error(self, int_frame_const_col):
         with pytest.raises(ValueError):
             df.apply(lambda x: Series([1, 2]), axis=1, result_type="broadcast")
 
-    def test_apply_raw(self, float_frame):
+    def test_apply_raw(self, float_frame, mixed_type_frame):
```
Comment (WillAyd) [inline pandas/tests/frame/test_apply.py]: Can you just make a separate test for this and reference GH issue number?

Comment (WillAyd) [reply pandas/tests/frame/test_apply.py]: Also can you assert that this is getting passed a 1D array?

**Review by WillAyd — CHANGES_REQUESTED** [review]
Thanks for the PR. Can you put the whatsnew note in for 1.1?

@jreback

**Review by jreback — CHANGES_REQUESTED** [review]
looks good. can you add a whatsnew note for 1.1, in bug fixes, reshaping.

File: pandas/tests/frame/test_apply.py [hunk 2]
```diff
@@ -250,6 +250,13 @@ def test_apply_raw(self, float_frame):
         expected = float_frame * 2
         tm.assert_frame_equal(result, expected)
 
+        # Mixed dtype
```
Comment (jreback) [inline pandas/tests/frame/test_apply.py]: can you move the _assert_raw to the top of this test & use it *also* on a single dtype frame (like you do below but on float_frame as well)

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by WillAyd — APPROVED** [review]
lgtm @jreback

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @kernc

**Pull request merged**